autogpt_platform/frontend/src/app/(platform)/build/components/NewBlockMenu/GraphSearchMenu/GraphSearchMenu.tsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { MagnifyingGlassIcon } from "@phosphor-icons/react";
+import { GraphSearchContent } from "./GraphSearchContent";
+import { ControlPanelButton } from "../ControlPanelButton";
+import { useGraphSearch } from "./useGraphSearch";
+import { CustomNode } from "@/components/CustomNode";
+
+interface GraphSearchMenuProps {
+  nodes: CustomNode[];
+  blockMenuSelected: "save" | "block" | "search" | "";
+  setBlockMenuSelected: React.Dispatch<
+    React.SetStateAction<"" | "save" | "block" | "search">
+  >;
+  onNodeSelect: (nodeId: string) => void;
+  onNodeHover?: (nodeId: string | null) => void;
+}
+
+export const GraphSearchMenu: React.FC<GraphSearchMenuProps> = ({
+  nodes,
+  blockMenuSelected,
+  setBlockMenuSelected,
+  onNodeSelect,
+  onNodeHover,
+}) => {
+  const { open, setOpen, searchQuery, setSearchQuery, filteredNodes } = 
+    useGraphSearch(nodes);
+
+  const handleNodeSelect = (nodeId: string) => {
+    onNodeSelect(nodeId);
+    setOpen(false);
+    setSearchQuery("");
+    setBlockMenuSelected("");
+  };
+
+  const handleOpenChange = (newOpen: boolean) => {
+    setOpen(newOpen);
+    setBlockMenuSelected(newOpen ? "search" : "");
+  };
+
+  return (
+    <Popover open={open} onOpenChange={handleOpenChange}>
+      <PopoverTrigger className="hover:cursor-pointer">
+        <ControlPanelButton
+          data-id="graph-search-control-popover-trigger"
+          data-testid="graph-search-control-button"
+          selected={blockMenuSelected === "search"}
+          className="rounded-none"
+        >
+          <MagnifyingGlassIcon className="h-5 w-5" weight="bold" />
+        </ControlPanelButton>
+      </PopoverTrigger>
+
+      <PopoverContent
+        side="right"
+        align="start"
+        alignOffset={-70}  // Offset upward to align with control panel top
```

Comment:
This doesn't seem to work and not sure how to make it appear aligned to the top of the control panel. @0ubbe  Do you know how to make this work